### DISCUSSION TITLE
So many questions i cant find answers for

### DISCUSSION THREAD

**Discussion by TeKett** [bmaltais/kohya_ss#1802] [Q&A]
The current ones on my mind is:

What is the actual resulting difference quality wise between using different batch sizes?

Should i or shouldn't i use a VAE? Preview images have better quality, but would this affect using a VAE when generating images later?

Number of cpu threads per core? Should this just match what i have? 2 with hyperthreading and 1 without? What happens if i set it to more then what i have? How does this affect the new consumer intel cards that only has hyperthreading on the first 8 and not the rest of the cores. I know that the cpu dont play a huge roll as compared to the gpu, so it might not even matter.

Do i need to include the lora when making the samples?

How does max token length work? If i use 75 on a caption 150 long does it discard the second half?

Can i train at multiple resolutions even if im only going to be generating images at a single one, or is this wasted training? As in should i make all training images the exact same size or not, or at least one of the directions the same.

When looking at bitsandbytes files i noticed it has files for cuda122, looking som more and cuda 123 was released September. Why hasn't the SD community upgraded from 118, considering 119 was released all the way back in april. Is there no benefit from using these versions? There's literally 0 related results when searching about it.

**Comment by zethfox** [comment]
Batch size to me seems bigger equals more generalized yet not as accurate... The bigger batch size I have to train slower or lose likeness... personally I use batch 2 gradient acclamation 2 for a nice balance of very accurate and flexible.

Can't comment on the vae... As I've never actually tried training with one

I'm not sure what the GPU threads do but if I set them to my maximum it slows me down so I set it to half of what my computer handles... On my 4090 I set it to 16

You don't need the Lora added to the sample, just add your trigger word... Where if you do keep tokens one it's the first token word before a comma...Markus, brown hair man, sitting... I would add Markus

Max tokens only need a boost if your captions txt are massive... To test your longest caption take it to auto 1111 if the token it returns is less than 75 you don't need to boost.

Yes you can train multiple resolution, buckets handle it for you. I find having fewer buckets is better...I like square 768x768, horizontal 1024x768 and vertical 768x1024..
Use birme.net to limit them 

The community hasn't upgraded from 11.8 because Facebook decided to stop providing binaries for any version of xformers past 11.8... You have to find or build the binaries yourself. Or do what I do and drop x formers completely and use Sdp or one of the others.
